reconcile/test/test_slack_base.py

```diff
@@ -0,0 +1,166 @@
+        "workspace": {
+            "name": "coreos",
+            "integrations": [
+                {
+                    "name": "dummy",
+                    "token": {
+                        "path": "some/path",
+                        "field": "bot_token",
+                    },
+                    "channel": "test",
+                    "icon_emoji": "test_emoji",
+                    "username": "foo"
+                },
+            ],
+            "api_client": {
+                "global": {
+                    "max_retries": 5,
+                    "timeout": 30
+                },
+                "methods": [
+                    {
+                        "name": "users.list",
+                        "args": "{\"limit\":123}"
+                    },
+                ]
+            }
+        }
+    }
+
+
+@pytest.fixture
+def unleash_slack_workspace():
+    return {
+        "workspace": {
+            "name": "coreos",
+            "integrations": [
+                {
+                    "name": "unleash-watcher",
+                    "token": {
+                        "path": "some/path",
+                        "field": "bot_token",
+                    }
+                },
+            ],
+            "api_client": {
+                "global": {
+                    "max_retries": 5,
+                    "timeout": 30
+                },
+                "methods": [
+                    {
+                        "name": "users.list",
+                        "args": "{\"limit\":123}"
+                    },
+                ]
+            }
+        },
+        "channel": "test",
+        "icon_emoji": "unleash",
+        "username": "test"
+    }
+
+
+@pytest.fixture
+def permissions_workspace():
+    return {
+        "workspace": {
+            "name": "coreos",
+            "token": {
+                "path": "some/path",
+                "field": "bot_token",
+            },
+            "api_client": {
+                "global": {
+                    "max_retries": 5,
+                    "timeout": 30
+                },
+                "methods": [
+                    {
+                        "name": "users.list",
+                        "args": "{\"limit\":123}"
+                    },
+                ]
+            },
+            "managedUsergroups": [
+                "foo"
+            ]
+        }
+    }
+
+
+def test_slack_workspace_raises():
+    with pytest.raises(ValueError):
+        slackapi_from_slack_workspace({}, {}, 'foo')
+
+
+def test_slack_workspace_ok(mocker, slack_workspace):
+    mock_rs = mocker.patch('reconcile.utils.secret_reader.SecretReader.read',
+                           return_value='secret')
+    mock_iu = mocker.patch(
+        'reconcile.utils.slack_api.SlackApi._initiate_usergroups')
+    a = slackapi_from_slack_workspace(slack_workspace, {}, 'dummy')
+    mock_rs.assert_called_once()
+    mock_iu.assert_called_once()
+
+    assert a.channel == 'test'
```

Comment:
I get you point, however the three fixtures differ not only in content, but also in the structure. It's due to subtle differences in the schema....

Reply:
That's fair, I reviewed very quickly and missed that the two that weren't related to permissions were a bit different. In this case, I'm not opinionated because there isn't duplication of fixtures (yet). You still have the problem of the values that are pulled from fixtures which aren't obvious from within the test. That is a bit more of a nit, however, but something to consider in the future.